slu/slu/utils/config.py

```diff
@@ -393,3 +375,49 @@ def remove_checkpoints(self, task) -> None:
 
     def get_supported_langauges(self) -> Dict[str, str]:
         return self._config["languages"]
+
+
+class OnStartupClientConfigDataProvider:
+
+    def __init__(self) -> None:
+        self.client_configs = {}
+
+    
+    def _is_valid_config_schema(self, config_dict):
+        #TODO: validate the incoming JSON
+        return True
+
+
+    def _parse_json(self, configs_response: List[Dict[str, Dict]]):
+
+        # if project_configs_response is of List[Dict]
+        for config_map in configs_response:
+            model_name = config_map.get(const.MODEL_NAME)
+            if self._is_valid_config_schema(config_map):
+                self.client_configs[model_name] = config_map
+
+
+    def _get_configs_from_api(self):
+
+        BUILDER_BACKEND_URL = os.getenv("BUILDER_BACKEND_URL")
+        if BUILDER_BACKEND_URL is None:
+            raise ValueError(
+                f"missing BUILDER_BACKEND_URL env variable, please set it appropriately."
+            )
+        url = BUILDER_BACKEND_URL + const.CLIENTS_CONFIGS_ROUTE
+        for _ in range(5):
```

Comment:
Requests have [code for retries](https://stackoverflow.com/questions/15431044/can-i-set-max-retries-for-requests-request).

Reply:
fixed it with `urllib3` based `Retry` along with `requests` based `HTTPAdapter`